server/src/lib/authentication/backends/ldap/LdapUsersDatabase.ts

```diff
@@ -85,10 +86,24 @@ export class LdapUsersDatabase implements IUsersDatabase {
       }
     )
     .catch((err) =>
-      Bluebird.reject(new Exceptions.LdapError("Failed during email retrieval: " + err.message))
+      Bluebird.reject(new Exceptions.LdapError("Failed during groups retrieval: " + err.message))
     );
   }
 
+  getUsersWithNetworkAddresses(): Bluebird<UsersWithNetworkAddresses[]> {
+    const that = this;
+    return that.withSession(
+      that.configuration.user,
+      that.configuration.password,
+      (session) => {
+        return session.searchWhitelist();
```

Comment:
Yeah this could potentially be somewhat taxing.
If/when attempting to do cache, I think it would be worthwhile to make how often it updates the cache a configurable parameter.